src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -16,25 +12,34 @@
 using System.Threading;
 using System.Threading.Tasks;
 using System.Xml;
-#if MANAGED_SNI
-using System.Data.SqlClient.SNI;
-#endif
+
 
 using MSS = Microsoft.SqlServer.Server;
 
 namespace System.Data.SqlClient
 {
+
+    struct SNIErrorDetails
+    {
+        public string errorMessage;
+        public uint nativeError;
+        public uint sniErrorNumber;
+        public int provider;
+        public uint lineNumber;
+        public string function;
+        public Exception exception;
+    }
+
     // The TdsParser Object controls reading/writing to the netlib, parsing the tds,
     // and surfacing objects to the user.
-    sealed internal class TdsParser
+    sealed internal partial class TdsParser
     {
         static TdsParser()
         {
             // For CoreCLR, we need to register the ANSI Code Page encoding provider before attempting to get an Encoding from a CodePage
             Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

Comment:
I am fine to track this as separate issue but I am still seeing the clinet consuming the library can decide about the encoding. if you have a client using Unicode only DBs why you want them pay for the encoding cost? this is similar to the networking libraries which is up to the consumer of the library to decide about the encoding but not forcing every app consume the library to pay for the encoding even if they are not using it or need it.